tests/transforms/test_zx.py

```diff
@@ -0,0 +1,654 @@
+
+        zx_g = qml.transforms.to_zx(qscript)
+
+        matrix_zx = zx_g.to_matrix()
+
+        assert isinstance(zx_g, pyzx.graph.graph_s.GraphS)
+        # Check whether the two matrices are each others conjugate transposes
+        mat_product = qml.math.dot(matrix_qscript, qml.math.conj(matrix_zx.T))
+        # Remove global phase
+        mat_product /= mat_product[0, 0]
+
+        assert qml.math.allclose(mat_product, I)
+
+        qscript_back = qml.transforms.from_zx(zx_g)
+        assert isinstance(qscript_back, qml.tape.QuantumScript)
+
+        matrix_qscript_back = qml.matrix(
+            qscript_back, wire_order=[i for i in range(0, len(qscript.wires))]
+        )
+
+        # Check whether the two matrices are each others conjugate transposes
+        mat_product = qml.math.dot(matrix_qscript, qml.math.conj(matrix_qscript_back.T))
+        # Remove global phase
+        mat_product /= mat_product[0, 0]
+
+        assert qml.math.allclose(mat_product, I)
+
+    @pytest.mark.parametrize("operation", non_diagram_like_operations)
+    def test_non_diagram_like_op(self, operation):
+        """Test operations that result in a non diagram like circuit."""
+
+        I = qml.math.eye(2 ** len(operation.wires))
+
+        qscript = QuantumScript([operation], [], [])
+        matrix_qscript = qml.matrix(qscript)
+
+        zx_g = qml.transforms.to_zx(qscript)
+        assert isinstance(zx_g, pyzx.graph.graph_s.GraphS)
+
+        matrix_zx = zx_g.to_matrix()
+        # Check whether the two matrices are each others conjugate transposes
+        mat_product = qml.math.dot(matrix_qscript, qml.math.conj(matrix_zx.T))
+        # Remove global phase
+        mat_product /= mat_product[0, 0]
+        assert qml.math.allclose(mat_product, I)
+
+        with pytest.raises(qml.QuantumFunctionError, match="Graph doesn't seem circuit like"):
+            qml.transforms.from_zx(zx_g)
+
+    @pytest.mark.parametrize("split", split)
+    @pytest.mark.parametrize("operation", non_diagram_like_operations)
+    def test_circuit(self, operation, split):
+        """Test a simple circuit."""
+
+        I = qml.math.eye(2**2)
+
+        operations = [
+            qml.RZ(5 / 4 * np.pi, wires=1),
+            qml.RZ(3 / 4 * np.pi, wires=1),
+            qml.RX(0.1, wires=0),
+            qml.PauliZ(wires=0),
+            qml.RZ(0.3, wires=1),
+            qml.PauliX(wires=1),
+            qml.CNOT(wires=[0, 1]),
+            qml.CNOT(wires=[1, 0]),
+            qml.SWAP(wires=[0, 1]),
+        ]
+
+        qscript = QuantumScript(operations, [], [])
+        zx_g = qml.transforms.to_zx(qscript)
+
+        assert isinstance(zx_g, pyzx.graph.graph_s.GraphS)
+
+        matrix_qscript = qml.matrix(qscript)
+        matrix_zx = zx_g.to_matrix()
+        # Check whether the two matrices are each others conjugate transposes
+        mat_product = qml.math.dot(matrix_qscript, qml.math.conj(matrix_zx.T))
+        # Remove global phase
```

Comment:
tangential, but I'm wondering if we should offer this as a convenience in PennyLane? maybe it's overkill :p

Reply:
Yes it can be a nice small project to check equivalence of circuit up to global phase, might be worth to add an issue 👍